eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/processors/SubSnpNoHgvsToDbsnpVariantsWrapperProcessorTest.java

```diff
@@ -51,13 +39,41 @@
 
     private static final Timestamp CREATED_DATE = Timestamp.valueOf("2001-01-05 12:30:50.0");
 
-    private SubSnpNoHgvsToSubmittedVariantProcessor processor;
+    private SubSnpNoHgvs subSnpNoHgvs;
+
+    private SubSnpNoHgvsToDbsnpVariantsWrapperProcessor processor;
 
     @Before
     public void setUp() throws Exception {
-        processor = new SubSnpNoHgvsToSubmittedVariantProcessor();
+        processor = new SubSnpNoHgvsToDbsnpVariantsWrapperProcessor();
+        subSnpNoHgvs = new SubSnpNoHgvs(25928972L, 14718243L, "A", "C", ASSEMBLY, BATCH_HANDLE,
+                                        BATCH_NAME, CHROMOSOME, CHROMOSOME_START, CONTIG_NAME,
+                                        CONTIG_START, DbsnpVariantType.SNV, Orientation.FORWARD,
+                                        Orientation.FORWARD, Orientation.FORWARD, false, false, false,
+                                        false, CREATED_DATE, CREATED_DATE, TAXONOMY);
     }
 
+    @Test
+    public void processSubmittedVariantAllelesMismatch() throws Exception {
+        DbsnpVariantsWrapper dbsnpVariantsWrapper = processor.process(subSnpNoHgvs);
+        assertSubmittedVariantDeclustered(dbsnpVariantsWrapper);
+        assertDeclusterOperationsRegistered(subSnpNoHgvs, dbsnpVariantsWrapper);
+        assertEquals(subSnpNoHgvs.getRsId(), dbsnpVariantsWrapper.getClusteredVariant().getAccession());
+    }
+
+    private void assertSubmittedVariantDeclustered(DbsnpVariantsWrapper dbsnpVariantsWrapper) {
+        assertEquals(1, dbsnpVariantsWrapper.getSubmittedVariants().size());
+        assertNull(dbsnpVariantsWrapper.getSubmittedVariants().get(0).getClusteredVariantAccession());
+    }
+
+    private void assertDeclusterOperationsRegistered(SubSnpNoHgvs subSnpNoHgvs,
+                                                     DbsnpVariantsWrapper dbsnpVariantsWrapper) {
+        assertEquals(1, dbsnpVariantsWrapper.getOperations().size());
+        assertEquals(EventType.UPDATED, dbsnpVariantsWrapper.getOperations().get(0).getEventType());
+        assertEquals(1, dbsnpVariantsWrapper.getOperations().get(0).getInactiveObjects().size());
+        assertEquals("Declustered (Alleles mismatch)", dbsnpVariantsWrapper.getOperations().get(0).getReason());
```

Comment:
If we add other decluster operations to this processor in the future, those method names should be improved (or add some parameter with the decluster reason)